oneandone/src/test/java/org/apache/jclouds/oneandone/rest/features/FirewallPolicyApiLiveTest.java

```diff
@@ -0,0 +1,188 @@
+
+      FirewallPolicy updateResult = firewallPolicyApi().assignServerIp(currentFirewallPolicy.id(), toAdd);
+
+      assertNotNull(updateResult);
+
+   }
+
+   @Test(dependsOnMethods = "testAssignServerIp")
+   public void testListServerIps() {
+      List<FirewallPolicy.ServerIp> servers = firewallPolicyApi().getServerIps(currentFirewallPolicy.id());
+
+      assertNotNull(servers);
+      assertFalse(servers.isEmpty());
+      Assert.assertTrue(servers.size() > 0);
+   }
+
+   @Test(dependsOnMethods = "testListServerIps")
+   public void testServerIpGet() {
+      FirewallPolicy.ServerIp result = firewallPolicyApi().getServerIp(currentFirewallPolicy.id(), currentServer.ips().get(0).id());
+
+      assertNotNull(result);
+   }
+
+   @Test(dependsOnMethods = "testServerIpGet")
+   public void testUnassignServer() {
+      FirewallPolicy result = firewallPolicyApi().unassignServerIp(currentFirewallPolicy.id(), currentServer.ips().get(0).id());
+
+      assertNotNull(result);
+      assertEquals(result.id(), currentFirewallPolicy.id());
+   }
+
+   @Test(dependsOnMethods = "testUnassignServer")
+   public void testAddRules() {
+      assertNodeAvailable(currentServer);
+      currentServer = updateServerStatus(currentServer);
+      List<FirewallPolicy.Rule.CreatePayload> rules = new ArrayList<FirewallPolicy.Rule.CreatePayload>();
+      FirewallPolicy.Rule.CreatePayload rule = FirewallPolicy.Rule.CreatePayload.builder()
+              .portFrom(4567)
+              .portTo(4567)
+              .protocol(Types.RuleProtocol.TCP)
+              .source("0.0.0.0")
+              .build();
+      rules.add(rule);
+      FirewallPolicy response = firewallPolicyApi().addRules(currentFirewallPolicy.id(), FirewallPolicy.Rule.AddRule.create(rules));
+
+      assertNotNull(response);
+
+   }
+
+   @Test(dependsOnMethods = "testAddRules")
+   public void testListRules() {
+      List<FirewallPolicy.Rule> servers = firewallPolicyApi().listRules(currentFirewallPolicy.id());
+
+      assertNotNull(servers);
+      assertFalse(servers.isEmpty());
+      Assert.assertTrue(servers.size() > 0);
+   }
+
+   @Test(dependsOnMethods = "testAddRules")
+   public void testGetRule() {
+      FirewallPolicy.Rule result = firewallPolicyApi().getRule(currentFirewallPolicy.id(), currentFirewallPolicy.rules().get(0).id());
+
+      assertNotNull(result);
+   }
+
+   @Test(dependsOnMethods = "testGetRule")
+   public void testRemoveRule() {
+      FirewallPolicy result = firewallPolicyApi().removeRule(currentFirewallPolicy.id(), currentFirewallPolicy.rules().get(0).id());
+
+      assertNotNull(result);
+      assertEquals(result.id(), currentFirewallPolicy.id());
+   }
+
+   @AfterClass(alwaysRun = true)
+   public void teardownTest() throws InterruptedException {
+      //give time for operations to finish
+      firewallPolicyApi().delete(currentFirewallPolicy.id());
+      Thread.sleep(15000);
```

Comment:
Doing this in a live test is a symptom of something that users will probably have to do themselves. better use the build-in `Predicates2.retry` construct to wait until the server is deleted and consider binding it to the guice context so it can be reused where needed.